### PR TITLE
restore service type input value on nav

### PR DESCRIPTION
**Pull request by rianfowler**
## Description
- restore initial input value for service type on navigation 

## Testing done
- verified locally by following steps to reproduce in linked [issue](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15418) 

## Acceptance criteria
- [x] steps to reproduce no longer manifest issue

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/facility-locator/components/ServiceTypeAhead.jsx [hunk 1]
```diff
@@ -56,12 +64,17 @@ class ServiceTypeAhead extends Component {
...
-    const { services } = this.state;
+    const { defaultSelectedItem, services } = this.state;
     // eslint-disable-next-line prettier/prettier
     const renderService = (s) => { return (s && s.name) ? s.name.trim() : ''; };
 
     return (
-      <Downshift onChange={this.handleOnSelect} itemToString={renderService}>
+      <Downshift
+        onChange={this.handleOnSelect}
+        defaultSelectedItem={defaultSelectedItem}
```
Comment (rianfowler) [inline src/applications/facility-locator/components/ServiceTypeAhead.jsx]: Was simple to find this once I realized we were two versions behind and I had to look at the [git history](https://github.com/paypal/downshift/blob/531bd8ea83ae5ecc3c163357ac9be4dc721d2060/README.md#defaultinputvalue) of Downshift's readme to find the right docs.

File: src/applications/facility-locator/components/ServiceTypeAhead.jsx [hunk 2]
```diff
@@ -56,12 +64,17 @@ class ServiceTypeAhead extends Component {
...
+        itemToString={renderService}
+        key={defaultSelectedItem}
```
Comment (rianfowler) [inline src/applications/facility-locator/components/ServiceTypeAhead.jsx]: Downshift sets a state off of its initial props and doesn't re-mount or reinitialize if those props change. The selected item is added to the redux state after the initialization so passing a key forces Downshift to re-mount if an initial state needs to be restored.

**Review by rianfowler — COMMENTED** [review]
I didn't see any tests to update for this.

**Comment by 1Copenut** [comment]
The core ask has been fixed with this PR. I have an open question to the design team about a new item, that is maybe best served as a separate ticket. When I try to clear out the Service Type input and search again without it, the value is remembered on blur:

screen shot 2019-01-09 at 2 06 17 pm

---
screen shot 2019-01-09 at 2 06 23 pm

**Comment by rianfowler** [comment]
@1Copenut Current implementation only updates the app state when something is selected. but we have the option to do something like clearing the selecting item when `onInputValueChange`. 

Like I mentioned in slack, we couple state of the search to the state of the input and we should de-couple that in the ux i.e. we should describe the list separately from the input. 

Couple of options: 
- given I've selected a service type and I've performed a search 
  - when I edit an input value and perform a new search without selecting an item from the service type list
  - then the input value should be restored to its previously selected value 
- given I've selected a service type and I've performed a search 
  - when I delete an input value and perform a new search without selecting an item from the service type list
  - then input value should remain empty and the selected value should revert to its default (all service types)

I think the focus behavior during the search for the above could follow something like this (this is basically how the school search works on the school feedback form):  
  - currently searching status e.g. "searching in Dallas, TX for Community Care (Non-VA Health) Facilities with all service types"
  - search results found e.g. "3 results found in Dallas, TX for Community Care (Non-VA Health) Facilities with Family Medicine Service Type"

We could utilize these areas for of the UI for these disclosures: 
screen shot 2019-01-09 at 14 56 38
screen shot 2019-01-09 at 14 56 29

We should roll these into a couple of separate issues after you've had a chance to circle back to UX.

**Comment by rianfowler** [comment]
cc @jaredcunha-usds @jbalboni 
re: the [above](https://github.com/department-of-veterans-affairs/vets-website/pull/9365#issuecomment-452884486) comment

This describes an autofill / search pattern we came up with on the school feedback form and that closely matches what's in the GI Bill Comparison Tool. If we wanted to have a standard recommendation for this, would we document this in the design system or in something else? 

It's come up a few times and I feel like people shouldn't have to keep reinventing this approach and going through a nearly identical 508 review.

**Comment by 1Copenut** [comment]
@rianfowler Yes, I wrote the guidance for the CMS autocomplete.

I agree there should be better messaging while we are searching, and when the search has returned. This is a good way to decouple the inputs from the results being shown.

I would also like to implement both of your suggestions for the typeahead search. These scenarios fit exactly with my mental model about how the input should behave as a user.

> Couple of options:
> * Given I've selected a service type and I've performed a search
>   * When I edit an input value and perform a new search without selecting an item from the service type list
>   * Then the input value should be restored to its previously selected value (like "Physical Therapy")
> * Given I've selected a service type and I've performed a search
>   * When I delete an input value and perform a new search without selecting an item from the service type list
>   * Then input value should remain empty and the selected value should revert to its default (all service types)

I'm going to create two new tickets for those updates, add design, and reference this one so we can track the work as part of a group.

**Comment by 1Copenut** [comment]
New issues added to track additional work appropriately:

* EPIC: [#16178](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16178)
* List and loading message: [#16177](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16177)
* Service type handling: [#16179](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16179)

**Review by 1Copenut — APPROVED** [review]
LGTM. I tested the branch locally and the values are being remembered after I run a search, go to the facility or provider detail, and return.

I opened a [few new issues](https://github.com/department-of-veterans-affairs/vets-website/pull/9365#issuecomment-453143347) linked in the comment above to track the comments made on this one, so we don't affect scope or effort here.

**Pull request merged**